doc/en/deprecations.rst

```diff
@@ -374,6 +374,36 @@ an appropriate period of deprecation has passed.
 
 Some breaking changes which could not be deprecated are also listed.
 
+.. _yield tests deprecated:
+
+``yield`` tests
+~~~~~~~~~~~~~~~
+
+.. versionremoved:: 8.4
```

Comment:
Ah, I think this is actually wrong - they were removed long ago, but in a way where the function was marked as xfail without running it; we just made it an error rather than an xfail now.  I don't know exactly when the transition was but I worry that this doc snippet might make users think that yield tests worked on 8.3, for example.

Reply:
Hmm good point, how about:

```suggestion
.. versionremoved:: 4.0

    ``yield`` tests ``xfail``.

.. versionremoved:: 8.4

    ``yield`` tests raise a collection error.
```

This is rendered like this:

![image](https://github.com/user-attachments/assets/d5c10556-5e6a-4162-b835-c6e4db0e56cc)

Reply:
That looks great.